CoinUtils/CMakeLists.txt

```diff
@@ -0,0 +1,169 @@
+  src/CoinPackedVector.cpp
+  src/CoinPackedVectorBase.cpp
+  src/CoinParam.cpp
+  src/CoinParamUtils.cpp
+  src/CoinPostsolveMatrix.cpp
+  src/CoinPrePostsolveMatrix.cpp
+  src/CoinPresolveDoubleton.cpp
+  src/CoinPresolveDual.cpp
+  src/CoinPresolveDupcol.cpp
+  src/CoinPresolveEmpty.cpp
+  src/CoinPresolveFixed.cpp
+  src/CoinPresolveForcing.cpp
+  src/CoinPresolveHelperFunctions.cpp
+  src/CoinPresolveImpliedFree.cpp
+  src/CoinPresolveIsolated.cpp
+  src/CoinPresolveMatrix.cpp
+  src/CoinPresolvePsdebug.cpp
+  src/CoinPresolveMonitor.cpp
+  src/CoinPresolveSingleton.cpp
+  src/CoinPresolveSubst.cpp
+  src/CoinPresolveTighten.cpp
+  src/CoinPresolveTripleton.cpp
+  src/CoinPresolveUseless.cpp
+  src/CoinPresolveZeros.cpp
+  src/CoinSearchTree.cpp
+  src/CoinShallowPackedVector.cpp
+  src/CoinSnapshot.cpp
+  src/CoinWarmStartBasis.cpp
+  src/CoinWarmStartVector.cpp
+  src/CoinWarmStartDual.cpp
+  src/CoinWarmStartPrimalDual.cpp
+  src/CoinRational.cpp)
+
+set(_HDRS
+  src/Coin_C_defines.h
+  src/CoinUtilsConfig.h
+  src/CoinAlloc.hpp
+  src/CoinBuild.hpp
+  src/CoinDenseVector.hpp
+  src/CoinDistance.hpp
+  src/CoinError.hpp
+  src/CoinFactorization.hpp
+  src/CoinSimpFactorization.hpp
+  src/CoinDenseFactorization.hpp
+  src/CoinOslFactorization.hpp
+  src/CoinFileIO.hpp
+  src/CoinFinite.hpp
+  src/CoinFloatEqual.hpp
+  src/CoinHelperFunctions.hpp
+  src/CoinIndexedVector.hpp
+  src/CoinLpIO.hpp
+  src/CoinMessage.hpp
+  src/CoinMessageHandler.hpp
+  src/CoinModel.hpp
+  src/CoinStructuredModel.hpp
+  src/CoinModelUseful.hpp
+  src/CoinMpsIO.hpp
+  src/CoinPackedMatrix.hpp
+  src/CoinPackedVector.hpp
+  src/CoinPackedVectorBase.hpp
+  src/CoinParam.hpp
+  src/CoinPragma.hpp
+  src/CoinPresolveDoubleton.hpp
+  src/CoinPresolveDual.hpp
+  src/CoinPresolveDupcol.hpp
+  src/CoinPresolveEmpty.hpp
+  src/CoinPresolveFixed.hpp
+  src/CoinPresolveForcing.hpp
+  src/CoinPresolveImpliedFree.hpp
+  src/CoinPresolveIsolated.hpp
+  src/CoinPresolveMatrix.hpp
+  src/CoinPresolveMonitor.hpp
+  src/CoinPresolvePsdebug.hpp
+  src/CoinPresolveSingleton.hpp
+  src/CoinPresolveSubst.hpp
+  src/CoinPresolveTighten.hpp
+  src/CoinPresolveTripleton.hpp
+  src/CoinPresolveUseless.hpp
+  src/CoinPresolveZeros.hpp
+  src/CoinSearchTree.hpp
+  src/CoinShallowPackedVector.hpp
+  src/CoinSignal.hpp
+  src/CoinSmartPtr.hpp
+  src/CoinSnapshot.hpp
+  src/CoinSort.hpp
+  src/CoinTime.hpp
+  src/CoinTypes.hpp
+  src/CoinUtility.hpp
+  src/CoinWarmStart.hpp
+  src/CoinWarmStartBasis.hpp
+  src/CoinWarmStartVector.hpp
+  src/CoinWarmStartDual.hpp
+  src/CoinWarmStartPrimalDual.hpp
+  src/CoinRational.hpp)
+
+add_library(CoinUtils ${_SRCS} ${_HDRS})
+target_include_directories(CoinUtils PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+  $<INSTALL_INTERFACE:include/coin>)
+target_compile_definitions(CoinUtils
+  PUBLIC HAVE_CONFIG_H
+  PRIVATE COINUTILS_BUILD)
+if(CMAKE_VERSION VERSION_LESS "3.8.2")
```

Comment:
Either this should be removed, or the toplevel `cmake_minimum_required(VERSION 3.15)` should be adjusted.